hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/sink/compact/HoodieFlinkCompactor.java

```diff
@@ -314,6 +314,9 @@ private void compact() throws Exception {
           .setParallelism(1);
 
       env.execute("flink_hudi_compaction_" + String.join(",", compactionInstantTimes));
+      if (conf.getBoolean(FlinkOptions.CLEAN_ASYNC_ENABLED) && conf.getBoolean(FlinkOptions.CLEAN_OFFLINE_ENABLE)){
+        writeClient.clean();
+      }
```

Comment:
`CompactionCommitSink#open` should already handle the cleaning.

Reply:
CompactionCommitSink#open only do clean when OptionsResolver.isInsertOverwrite(conf) is true.

  @Override
  public void open(Configuration parameters) throws Exception {
    super.open(parameters);
    if (conf.getBoolean(FlinkOptions.CLEAN_ASYNC_ENABLED)) {
      this.writeClient = FlinkWriteClients.createWriteClient(conf, getRuntimeContext());
      this.executor = NonThrownExecutor.builder(LOG).waitForTasksFinish(true).build();

      if (OptionsResolver.isInsertOverwrite(conf)) {
        String instantTime = HoodieActiveTimeline.createNewInstantTime();
        LOG.info(String.format("exec sync clean with instant time %s...", instantTime));
        executor.execute(() -> writeClient.clean(instantTime), "wait for sync cleaning finish");
      }
    }
  }